packages/job-worker/src/playout/upgrade.ts

```diff
@@ -67,13 +69,36 @@ export async function handleBlueprintUpgradeForStudio(context: JobContext, _data
 			}),
 		])
 	)
+	const routeSets = Object.fromEntries(
+		Object.entries<unknown>(result.routeSets ?? {}).map((dev) => [
+			dev[0],
+			literal<Complete<StudioRouteSet>>({
+				name: '', //dev[1]?.name ?? '',
+				active: false, // dev[1].options ?? {},
+				defaultActive: false, //dev[1].defaultActive ?? false,
+				behavior: 1, //dev[1].behavior ?? {},
+				exclusivityGroup: undefined, //dev[1].exclusivityGroup ?? undefined,
+				routes: [], //dev[1].routes,
+			}),
+		])
+	)
+	const routeSetExclusivityGroups = Object.fromEntries(
+		Object.entries<unknown>(result.routeSetExclusivityGroups ?? {}).map((dev) => [
+			dev[0],
+			literal<Complete<StudioRouteSetExclusivityGroup>>({
+				name: '', //dev[1].name,
+			}),
+		])
+	)
 
 	await context.directCollections.Studios.update(context.studioId, {
 		$set: {
 			'mappingsWithOverrides.defaults': translateMappings(result.mappings),
 			'peripheralDeviceSettings.playoutDevices.defaults': playoutDevices,
 			'peripheralDeviceSettings.ingestDevices.defaults': ingestDevices,
 			'peripheralDeviceSettings.inputDevices.defaults': inputDevices,
+			'settings.routeSets': routeSets,
+			'settings.routeSetExclusivityGroups': routeSetExclusivityGroups,
```

Comment:
These properties on the studio need to be converted to use `ObjectWithOverrides<...>`.

As it is currently implemented, every time the blueprint upgrade scripts are re-run, it will replace all the existing route-sets in sofie with whatever the blueprints have defined.
This is behaviour is hostile to the user, as it will remove any changes they have made. The `ObjectWithOverrides` type and flow was created to allow for this blob generation by blueprints while allowing the user to override portions and for those overrides to be persisted.

Glancing at my nrk hat, I have some routesets setup on just my machine to allow me to test some things in my limited setup. I need these to not be defined via blueprints, which I also do with mappings and works because of the `ObjectWithOverrides` structure being used there.